Kudu.Core/SourceControl/Git/LibGit2SharpRepository.cs

```diff
@@ -0,0 +1,399 @@
+
+                        string content = @"#!/bin/sh
+if [ " + "\"$1\" = \"get\"" + @" ]; then
+      echo username=dummyUser
+      echo password=dummyPassword
+fi" + "\n";
+
+                        File.WriteAllText(GitCredentialHookPath, content);
+
+                        repo.Config.Set("credential.helper", string.Format("!'{0}'", GitCredentialHookPath));
+                    }
+                }
+                using (tracer.Step("Setup post receive hook"))
+                {
+                    FileSystemHelpers.EnsureDirectory(Path.GetDirectoryName(PostReceiveHookPath));
+
+                    string content = @"#!/bin/sh
+read i
+echo $i > pushinfo
+" + KnownEnvironment.KUDUCOMMAND + "\n";
+
+                    File.WriteAllText(PostReceiveHookPath, content);
+                }
+
+                // NOTE: don't add any new init steps after creating the post receive hook,
+                // as it's also used to mark that Init was fully executed
+            }
+        }
+
+        private string GitCredentialHookPath
+        {
+            get
+            {
+                return Path.Combine(RepositoryPath, ".git", "hooks", "git-credential-invalid.sh");
+            }
+        }
+
+        public ChangeSet GetChangeSet(string id)
+        {
+            using (var repo = new LibGit2Sharp.Repository(RepositoryPath))
+            {
+                LibGit2Sharp.Commit commit = repo.Lookup<LibGit2Sharp.Commit>(id);
+
+                if (commit == null)
+                {
+                    return null;
+                }
+
+                return new ChangeSet(commit.Sha, commit.Author.Name, commit.Author.Email, commit.Message, commit.Author.When);
+            }
+        }
+
+        public void AddFile(string path)
+        {
+            using (var repo = new LibGit2Sharp.Repository(RepositoryPath))
+            {
+                repo.Stage(path);
+            }
+        }
+
+        public bool Commit(string message, string authorName, string emailAddress)
+        {
+            using (var repo = new LibGit2Sharp.Repository(RepositoryPath))
+            {
+                try
+                {
+                    var changes = repo.RetrieveStatus().Select(c => c.FilePath);
+                    if (!changes.Any())
+                    {
+                        return false;
+                    }
+
+                    repo.Stage(changes);
+                    if (string.IsNullOrEmpty(authorName) ||
+                        string.IsNullOrEmpty(emailAddress))
+                    {
+                        repo.Commit(message);
+                    }
+                    else
+                    {
+                        repo.Commit(message, new Signature(authorName, emailAddress, DateTimeOffset.UtcNow));
+                    }
+                    return true;
+                }
+                catch(LibGit2Sharp.EmptyCommitException)
```

Comment:
Now that you exit early when no changes have been detected, do we still need this `catch` clause?